shared/nav.desktop.js

```diff
@@ -56,59 +57,85 @@ class Nav extends Component {
     // If you startup while offline, you'll stay in an errored state
     window.addEventListener('online', () => this.props.bootstrap())
 
-    this._renderedActiveTab = null // the last tab we actually drew
+    this._lastCheckedTab = null // the last tab we resized for
   }
 
   _handleTabsChange (e, key, payload) {
     this.props.switchTab(e)
   }
 
-  _resizeWindow () {
-    this._askedResize = false
-    const currentWindow = remote.getCurrentWindow()
-
-    if (!currentWindow) {
+  _checkTabChanged () {
+    if (this._checkingTab) {
       return
     }
 
-    if (this._renderedActiveTab === loginTab) {
-      const [width, height] = currentWindow.getContentSize()
+    this._checkingTab = true
+
+    setImmediate(() => {
+      this._checkingTab = false
```

Comment:
this is all single threaded and nothing in here calls back into checkTabChanged. i'm just worried about some logic below not setting it and then we're stuck (like an early return, of which there are some below)